### PR TITLE
Accept a single or multiple inputs to _inference

### PR DESCRIPTION
**Pull request by davidkyle**
Enhances the _inference API to take either a single input or multiple inputs:

```
POST _inference/sparse_embedding/my-elser-model
{
  "input": "a single input"
}
```

```
POST _inference/sparse_embedding/my-elser-model
{
  "input": ["or", "mutltiple", "inputs"]
}
```

The response will always be an array of results:
```
{
  "sparse_embedding": [
    {
      "port": 2.1259406,
      "sky": 1.7073475,
       (...)
    }
    (...)
  ]
}
```

### CODE REVIEW

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/huggingface/HuggingFaceElserResponseEntity.java [hunk 1]
```diff
@@ -70,11 +70,11 @@ public static TextExpansionResults fromResponse(HttpResult response) throws IOEx
             );
 
             if (parsedResponse.isEmpty()) {
-                return new TextExpansionResults(DEFAULT_RESULTS_FIELD, Collections.emptyList(), false);
+                return List.of(new TextExpansionResults(DEFAULT_RESULTS_FIELD, Collections.emptyList(), false));
             }
 
             // we only handle a single response right now so just grab the first one
-            return parsedResponse.get(0);
```
Comment (jonathan-buttner) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/huggingface/HuggingFaceElserResponseEntity.java:77]: 💯

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/huggingface/HuggingFaceElserActionTests.java [hunk 2]
```diff
@@ -66,6 +69,7 @@ public void shutdown() throws IOException {
         webServer.close();
     }
 
+    @SuppressWarnings("unchecked")
```
Comment (jonathan-buttner) [inline x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/huggingface/HuggingFaceElserActionTests.java:72]: Just curious is this for the use of generics in the `PlainActionFuture`?

**Review by jonathan-buttner — APPROVED** [review]

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/huggingface/HuggingFaceElserActionTests.java (hunk 2 above)
Comment (davidkyle) [reply x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/huggingface/HuggingFaceElserActionTests.java:72]: No there is a (List<String>) cast in the test

**Comment by davidkyle** [comment]
@elasticmachine test this please

**Pull request merged**